### PR TITLE
ES nested fields autocomplete

### LINKED ISSUE #4526: [Feature Request] getFields() doesn't support nested mapping; fails if current date's index is absent
in the panel graph editor mode, while trying to select fields for example, under Metric 'Average', if an elasticsearch document is nested, then the autocomplete feature doesn't show up. 

Also, I've observed that on debugging, when the index name using a daily pattern, doesn't have "today's date" in it, getFields feature fails to run smooth, since  _mapping for the index to which my datasource belongs, is not present..

Would be nice to have this sorted out. 

--
* What grafana version are you using?
3.0.0 (latest from master branch)

* What datasource are you using?
elasticsearch

* What OS are you running grafana on?
fedora 23

### PR DESCRIPTION
**Pull request by alexanderzobnin**
This PR based on #4527 and fixes #4526 - show nested fields in ES autocomplete.

### CODE REVIEW

File: public/app/plugins/datasource/elasticsearch/datasource.js [hunk 1]
```diff
@@ -47,10 +47,19 @@ function (angular, _, moment, kbn, ElasticQueryBuilder, IndexPattern, ElasticRes
     };
 
     this._get = function(url) {
-      return this._request('GET', this.indexPattern.getIndexForToday() + url).then(function(results) {
-        results.data.$$config = results.config;
-        return results.data;
-      });
+      var range = timeSrv.timeRange();
+      var index_list = this.indexPattern.getIndexList(range.from.valueOf(), range.to.valueOf());
```
Comment (torkelo) [inline public/app/plugins/datasource/elasticsearch/datasource.js]: What is this change doing?

Comment (alexanderzobnin) [inline public/app/plugins/datasource/elasticsearch/datasource.js]: Seems, he wanted to use dashboard's time range for getting right index. For example, if you retrieve data for previous day and this index has another mapping.

Comment (arcolife) [inline public/app/plugins/datasource/elasticsearch/datasource.js]: yes, exactly.

Comment (torkelo) [inline public/app/plugins/datasource/elasticsearch/datasource.js]: maybe that should be in another PR?

Comment (alexanderzobnin) [inline public/app/plugins/datasource/elasticsearch/datasource.js]: I think, autocomplete may not work properly without this change. For example, you haven't index for today, but just want to look at data for previous week. In this case `getIndexForToday()` returns not existed index name.

Comment (torkelo) [inline public/app/plugins/datasource/elasticsearch/datasource.js]: I see

**Comment by Mormaii** [comment]
Would this support wildcards on nested fields too? Such as graphing "disk.*"

**Comment by alexanderzobnin** [comment]
@Mormaii not, this PR isn't implements this.
